src/Eloquent/HybridRelations.php

```diff
@@ -249,13 +251,18 @@ public function belongsToMany(
             );
         }
 
+
         // First, we'll need to determine the foreign key and "other key" for the
         // relationship. Once we have determined the keys we'll make the query
         // instances as well as the relationship instances we need for this.
         $foreignKey = $foreignKey ?: $this->getForeignKey().'s';
 
         $instance = new $related;
 
+        if ($otherKey === $relation) {
+            throw new \LogicException(sprintf('In %s::%s(), the key cannot be identical to the relation name "%s". The default key is "%s".', static::class, $relation, $relation, $instance->getForeignKey().'s'));
```

Comment:
@mshamaseen [wrote](https://github.com/mongodb/laravel-mongodb/pull/2577):
> The solution was so easy, just don't name the foreign key the same as the relation; that is the default behavior if you don't pass any foreign keys to $this->belongsToMany(related , /* Don't pass anything else */) , and that is also the documented way to do it.

I propose to throw an exception in this case to avoid any error.